Assets/_Laz/Editor/PatrolLocationHandler.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using UnityEditor;
+using Laz;
+using UnityEngine;
+
+
+namespace LazEditor
+{
+    [CustomEditor(typeof(PatrolBehaviour))]
+    public class PatrolLocationHandler : Editor
+    {
+        private const float DRAW_PLANE_HEIGHT = 0f;
+        private const float CLOSEST_DISTANCE_FOR_DELETION = 0.6f;
+        private PatrolBehaviour _behaviour = null;
+        private Vector3 _tempPosition = Vector3.zero;
+        private List<Vector3> _patrolTrail;
+
+        private void OnEnable()
+        {
+            _behaviour = (PatrolBehaviour) target;
```

Comment:
so what happens if target is null? 

and what is target?

Reply:
So target is just the PatrolBehaviour Script that the editor is trying to access. I think normally it cant be null. but I put a null check regardless in OnSceneGUI